runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -273,22 +277,52 @@ void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace
      * @param timestamp        if specified, all stream updates from this timestamp will be returned
      *                         if null, only future updates will be returned
      */
-    public void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace,
-                   @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
-                   @Nullable Timestamp timestamp) {
-        runtime.getTableRegistry().getStreamManager()
+    public void subscribeListener(@Nonnull StreamListener streamListener, @Nonnull String namespace,
+                                  @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                                  @Nullable Timestamp timestamp) {
+        runtime.getTableRegistry().getStreamingManager()
                 .subscribe(streamListener, namespace, streamTag, tablesOfInterest,
                         (timestamp == null) ? getTimestamp().getSequence() : timestamp.getSequence());
     }
 
+    /**
+     * Subscribe to transaction updates on specific tables with the streamTag in the namespace.
+     * Objects returned will honor transactional boundaries.
+     *
+     * @param streamListener   client listener for callback
+     * @param namespace        the CorfuStore namespace to subscribe to
+     * @param streamTag        only updates of tables with the stream tag will be polled
+     * @param tablesOfInterest only updates from these tables of interest will be sent to listener
+     * @param timestamp        if specified, all stream updates after this timestamp will be returned
+     *                         if null, only future updates will be returned
+     * @param bufferSize       maximum size of buffered transaction entries
+     */
+    public void subscribeListener(@Nonnull StreamListener streamListener, @Nonnull String namespace,
```

Comment:
Do you need to expose this to the shim `CorfuStoreShim` ?

Reply:
Done.